### ISSUE TITLE
[BUG] csv_reader segfaults reading gzipped file

### ISSUE THREAD

**Issue by beckernick** [rapidsai/cudf#3379]
I’m unable to read a gzipped file with cuDF’s csv reader, but pandas succeeds. Getting the following error from a nightly dev container `rapidsai/rapidsai-dev-nightly:0.11-cuda10.1-runtime-ubuntu18.04-py3.7` from 11/12/19. Both libraries can read the file if it is already decompressed.

This file causes a segfault, rather than a Runtime error as in #3378 .

Details on slack.

Causes  segfault/core dump:
```python
cols = [...]
usecols = [...]
df = cudf.read_csv("/path/to/data.dat.gz",
                   compression="gzip",
                   sep="|",
                   names=cols,
                   header=None,
                   usecols=usecols
                  )

```

Pandas:
```python
import pandas as pd
cols = [...]
usecols = [...]
df = pd.read_csv("/path/to/data.dat.gz",
                   compression="gzip",
                   sep="|",
                   names=cols,
                   header=None,
                   usecols=usecols
                  )
print(df.shape)
(248197173, 4)
```

**Comment by OlivierNV** [comment]
Is this 4 columns, 250M rows ? (was the file greater than 4GB ?)

**Comment by vuule** [comment]
Assuming that the compressed file size is larger than 4GB, the source size (32 bit signed integer) would overflow and getUncompressedHostData() would get a negative size as the input size parameter.
I don't have the repro file, but have verified that passing a negative source size to getUncompressedHostData() results in segfault. 
We can easily check for this in getUncompressedHostData() and avoid the segfault, but it would be even better to enable large compressed input. @OlivierNV, is there a reason why source size is a 32 bit value?

**Comment by OlivierNV** [comment]
Yeah, not sure when this happened, but src_size should be size_t type and *not* cudf::size_type, that's a regression ~ not sure when this happened.

**Comment by vuule** [comment]
Yeah, size_t is used everywhere else in the call stack.
Made the change already: https://github.com/rapidsai/cudf/pull/3775

**Comment by vuule** [comment]
Issue should be fixed with #3775. @beckernick , can you please verify that you can open the file now?

**Comment by kkraus14** [comment]
Fixed by #3775 closing.
